### PR TITLE
Switch to github action

### PR DESCRIPTION
**Pull request by hunterlxt**
Travis has wasted a lot of our time, maybe we should try something new.

### CODE REVIEW

**Comment by hunterlxt** [comment]
Seems that everything is fine. https://github.com/hunterlxt/grpc-rs/actions/runs/75753983

**Comment by hunterlxt** [comment]
Looks like travis "refuses" to pass this pull request LOL @BusyJay

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    env:
+      RUST_BACKTRACE: 1
+      RUSTFLAGS: "--deny=warnings"
+    steps:
+    - uses: actions/checkout@v2
+    - run: sudo apt-get install -y clang-tidy-9
+    - run: sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-9 100
+    - run: which go && go version && which cargo && cargo version && clang --version && openssl version
+    - run: scripts/reset-submodule.cmd
+    - run: cargo fmt --all -- --check
```
Comment (BusyJay) [inline .github/workflows/ci.yml:22]: All these checks can be separated into a single job.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    - run: scripts/lint-grpc-sys.sh && git diff-index --quiet HEAD
+    - run: scripts/generate-bindings.sh && git diff --exit-code HEAD
+    - run: cargo clippy --all -- -D clippy::all && cargo clippy --all --no-default-features --features prost-codec -- -D clippy::all
+    - run: cargo build --no-default-features --features protobuf-codec
+    - run: cargo build --no-default-features --features prost-codec
+    - run: cargo build --all
+    - run: cargo test --all
+    # TODO: can't find library grpc via pkg-config in CI environment
+    # - run: GRPCIO_SYS_USE_PKG_CONFIG=1 RUSTFLAGS="-A unused-attributes" cargo test --all
```
Comment (BusyJay) [inline .github/workflows/ci.yml:31]: dynamic and static link can be separated.

**Comment by hunterlxt** [comment]
https://github.com/hunterlxt/grpc-rs/actions/runs/77042717 PTAL @BusyJay

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  Linux-Stable:
+    name: Linux-Stable
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: which go && go version && which cargo && cargo version && clang --version && openssl version
+    - run: scripts/reset-submodule.cmd
+    - run: scripts/generate-bindings.sh && git diff --exit-code HEAD
+    - run: cargo build --no-default-features --features protobuf-codec
```
Comment (BusyJay) [inline .github/workflows/ci.yml:41]: Should check `cargo build --no-default-features`.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    - run: cargo build --no-default-features --features prost-codec
+    - run: cargo build --all
```
Comment (BusyJay) [inline .github/workflows/ci.yml:43]: Should not use `--all`.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    - run: cargo test --all
+
+  Linux-Stable-Security:
```
Comment (BusyJay) [inline .github/workflows/ci.yml:46]: s/Security/openssl/

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    - run: rustup default nightly
+    - run: which go && go version && which cargo && cargo version && clang --version && openssl version
+    - run: scripts/reset-submodule.cmd
+    - run: cargo build --no-default-features --features protobuf-codec
+    - run: cargo build --no-default-features --features prost-codec
+    - run: cargo build
+    - run: cargo test --all
+    - run: RUSTFLAGS="-Z sanitizer=address" cargo test --all --target x86_64-unknown-linux-gnu
+
+  Linux-Nightly-Security:
```
Comment (BusyJay) [inline .github/workflows/ci.yml:70]: Just stable is enough.

**Pull request merged**